src/Extensions/BlockArchiveExtension.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace SilverStripe\VersionedAdmin\Extensions;
+
+use DNADesign\Elemental\Models\BaseElement;
+use SilverStripe\Forms\GridField\GridFieldDataColumns;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\Security\Member;
+use SilverStripe\VersionedAdmin\ArchiveAdmin;
+use SilverStripe\VersionedAdmin\ArchiveViewProvider;
+
+/**
+ * Adds a archive view for Elemental blocks
+ */
+class BlockArchiveExtension extends DataExtension implements ArchiveViewProvider
+{
+    /**
+     * @inheritDoc
+    */
+    public function getArchiveFieldClass()
+    {
+        return BaseElement::class;
+    }
+
+    /**
+     * @inheritDoc
+    */
+    public function getArchiveField()
+    {
+        $listField = ArchiveAdmin::createArchiveGridField('Blocks', BaseElement::class);
+
+        $listColumns = $listField->getConfig()->getComponentByType(GridFieldDataColumns::class);
+        $listColumns->setDisplayFields([
+            'Title' => BaseElement::singleton()->fieldLabel('Title'),
+            'Type' => _t(ArchiveAdmin::class . '.COLUMN_TYPE', 'Type'),
+            'LastEdited.Ago' => _t(ArchiveAdmin::class . '.COLUMN_DATEARCHIVED', 'Date Archived'),
+            'Breadcrumbs' => _t(ArchiveAdmin::class . '.COLUMN_ORIGIN', 'Origin'),
+            'AuthorID' => _t(ArchiveAdmin::class . '.COLUMN_ARCHIVEDBY', 'Archived By')
+        ]);
+        $listColumns->setFieldFormatting([
+            'Breadcrumbs' => function ($val, $item) {
+                $parent = $item->Page;
+
+                return $parent ? $parent->Breadcrumbs() : null;
+            },
+            'AuthorID' => function ($val, $item) {
+                return Member::get_by_id($val) ?: Member::get_by_id($val)->Name;
```

Comment:
That's not going to work. The Elvis Operator return the left side argument if it's *truthy* and the right side argument otherwise.

If your Member exists, the actual Member Object will be return rather than their Name.
If your Member doesn't exist, it will try to fetch it again and throw an Exception.

Also, you're doing 2 DB calls to fetch a simple bit of information..

Reply:
![image](https://user-images.githubusercontent.com/1168676/42609635-97b5008e-85e1-11e8-99b2-a49bef4bdd9e.png)